src/e2e-test/features/spanner/sink/BigQueryToSpanner_WithMacro.feature

```diff
@@ -0,0 +1,64 @@
+    Then Enter Spanner property "primaryKey" as macro argument "spannerSinkPrimaryKey"
+    Then Enter Spanner cmek property "encryptionKeyName" as macro argument "cmekSpanner" if cmek is enabled
+    Then Validate "Spanner" plugin properties
+    Then Close the Spanner properties
+    Then Connect source as "BigQuery" and sink as "Spanner" to establish connection
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "projectId" for key "bqProjectId"
+    Then Enter runtime argument value "projectId" for key "bqDatasetProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "bqServiceAccount"
+    Then Enter runtime argument value "dataset" for key "bqDataset"
+    Then Enter runtime argument value for BigQuery source table name key "bqSourceTable"
+    Then Enter runtime argument value "projectId" for key "spannerProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "spannerServiceAccount"
+    Then Enter runtime argument value for Spanner Instance ID key "spannerInstanceId"
+    Then Enter runtime argument value for Spanner Database Name key "spannerDatabaseName"
+    Then Enter runtime argument value for Spanner Sink Table Name key "spannerTablename"
+    Then Enter runtime argument value "spannerSinkPrimaryKeyBQ" for key "spannerSinkPrimaryKey"
+    Then Enter runtime argument value "cmekSpanner" for Spanner cmek property key "cmekSpanner" if Spanner cmek is enabled
+    Then Run the preview of pipeline with runtime arguments
+    Then Verify the preview of pipeline is "success"
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "bqProjectId"
+    Then Enter runtime argument value "projectId" for key "bqDatasetProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "bqServiceAccount"
+    Then Enter runtime argument value "dataset" for key "bqDataset"
+    Then Enter runtime argument value for BigQuery source table name key "bqSourceTable"
+    Then Enter runtime argument value "projectId" for key "spannerProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "spannerServiceAccount"
+    Then Enter runtime argument value for Spanner Instance ID key "spannerInstanceId"
+    Then Enter runtime argument value for Spanner Database Name key "spannerDatabaseName"
+    Then Enter runtime argument value for Spanner Sink Table Name key "spannerTablename"
+    Then Enter runtime argument value "spannerSinkPrimaryKeyBQ" for key "spannerSinkPrimaryKey"
+    Then Enter runtime argument value "cmekSpanner" for Spanner cmek property key "cmekSpanner" if Spanner cmek is enabled
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
```

Comment:
Add validation step 
Then Validate records transferred to target spanner table with record counts of BigQuery table